easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -595,8 +596,17 @@ def update(self, key, value, allow_duplicate=True):
             for item in lval:
                 if allow_duplicate or item not in param_value:
                     param_value = param_value + [item]
+        elif isinstance(param_value, tuple):
+            param_value = list(param_value)
+            for item in lval:
+                if allow_duplicate or item not in param_value:
+                    param_value.append(item)
+            param_value = tuple(param_value)
```

Comment:
This could actually be collapsed with the case above, since this approach works both for `list` and `tuple` values.

This line:

```python
param_value = list(param_value)
```

will result in a copy of the original in case of a `list` value, to which you can safely `.append` (we don't do that in the code above to avoid changing the original `param_value` in place.

```python
elif isinstance(param_value, (list, tuple)):
    param_value = list(param_value)
    ...

Reply:
Nice!  Collapsing if blocks is something I really like to do in my own code.

Reply:
See my updated comment below.  I think ([line 589](https://github.com/kelseymh/easybuild-framework/blob/4fd2205a007f37ac2efbd713636fa206fa4a7e1a/easybuild/framework/easyconfig/easyconfig.py#L589) that param_value is already a local copy that the function manipulates, then assigns back into `self[key]`.